libraries/AP_Math/vector2.h

```diff
@@ -89,6 +89,11 @@ struct Vector2
     // dot product
     T operator *(const Vector2<T> &v) const;
 
+    // dot product for Lua
```

Comment:
Not sure if this is supposed to be here?

Reply:
Indeed.. I hadn't noticed that there's a method directly above that can be used instead.  Copter is using Vector3f::dot() so I basically just copy-pasted that but now I see Copter could also be changed to use operator*.   I suspect thought that Leonard would prefer to stick with "::dot()" because it's more clear than reusing "operator*()".

.. I don't feel strongly about it but I think I might just fix the comment above Vector2f::dot() but keep using it although it's slightly wasteful to have two methods that do the same thing.. ?

Reply:
I also think using the `dot` method is easier to follow. But there is value in keeping the code base consistent. Maybe we can change them all....